README.md

###Tehran Linux Users Group Website

---
####How to edit and create new entry on website
Entries are on contents/entries as php files and you can edit them.
To create new entry you can use issue_session.sh from scripts directory.


ChangeLog
2015-02-15  Mohammad Reza Kamalifard <mr.kamalifard@gmail.com>

  * scripts/next_session.date, scripts/next_session.id: change next session
  info

2015-01-20  Ashkan Ghassemi <ghassemi@ftml.net>

  * contents/entries/209.php, scripts/next_session.date,
  scripts/next_session.id: 209 added.

2014-12-08  Ashkan Ghassemi <ghassemi@ftml.net>

  * contents/entries/207.php, contents/entries/208.php,
  scripts/next_session.date, scripts/next_session.id: 208 added. 207 updated.

2014-11-23  Ashkan Ghassemi <ghassemi@ftml.net>

  * images/tehlug-logo.png, menu.php: github account added. menu items
  shortened. tehlug logo added to source tree.

  * contents/calendar.php, contents/entries/207.php, menu.php,
  scripts/next_session.date, scripts/next_session.id: 207 added. 206 not
  updated yet. Trello public account added.

2014-11-09  Ashkan Ghassemi <ghassemi@ftml.net>

  * contents/entries/205.php, contents/entries/206.php,
  scripts/next_session.date, scripts/next_session.id: 205 updated. 206 added.

2014-10-07  Ashkan Ghassemi <ghassemi@ftml.net>

  * contents/entries/205.php, scripts/next_session.date,
  scripts/next_session.id: 205 added.

2014-09-29  Ashkan Ghassemi <ghassemi@ftml.net>

  * contents/front.php, images/success-of-opensource.jpe: front page updated:
  The Success of Open Source by Steven Weber.

2014-09-23  Ashkan Ghassemi <ghassemi@ftml.net>

  * contents/entries/203.php, contents/entries/204.php,
  scripts/issue_session.sh, scripts/next_session.date, scripts/next_session.id,
  style.css: 203 updated. 204 added. css changed to reflect member lists a in
  triple column view.

2014-09-09  Ashkan Ghassemi <ghassemi@ftml.net>

  * contents/entries/202.php, contents/entries/203.php, contents/faqs.php,
  contents/front.php, images/map-tosan-armaghan.png, style.css: 203 added. 202
  updated. (Still requires name entry, pictures, presentation list, etc.) some
  minor fixes to FAQ. New map (Tosan at Armaghan) added to front page.

2014-07-09  Behnam Tavakkoli kermani <b.tavakkoli@gmail.com>

  * contents/entries/201.php, contents/entries/202.php: ezafe kardan asami 201
  va khabare 202

2014-07-03  Ashkan Ghassemi <ghassemi@ftml.net>

  * ChangeLog, contents/entries/200.php, functions.php,
  scripts/gitlog2changelog: quick fix for & in HTML: 200.php line 24. must be
  &amp; instead.

2014-07-02  Behnam Tavakkoli kermani <b.tavakkoli@gmail.com>

  * contents/entries/200.php, photos/200/tehlug-200-thumb.jpg,
  photos/200/tehlug-200.jpg: eslahe site va darje khabare 201

  * contents/entries/201.php, contents/front.php, style.css: eslahe site va
  darje khabare 201

2014-06-21  Behnam Tavakkoli kermani <b.tavakkoli@gmail.com>

  * scripts/gitlog2changelog, scripts/issue_session.sh,
  scripts/make_thumb_photos.sh: darje 200 unknown

  * contents/entries/199.php, contents/entries/200.php, contents/front.php,
  images/map-hozehonari.jpg, images/tehlug-200-poster.jpg, style.css: ezafe
  kardan khabare 200 va asami 199

2014-06-07  Behnam Tavakkoli kermani <b.tavakkoli@gmail.com>

  * contents/entries/198.php, contents/entries/199.php: darje khabar 199 va
  asami 198

2014-05-30  Behnam Tavakkoli kermani <b.tavakkoli@gmail.com>

  * contents/entries/197.php, contents/entries/198.php: darje asami 197 +
  khabare 198

2014-05-15  Behnam Tavakkoli kermani <b.tavakkoli@gmail.com>

  * contents/entries/195.php, contents/entries/196.php,
  contents/entries/197.php: update jalaase 195 & 196 va darje khabare 197

2014-04-09  Behnam Tavakkoli kermani <b.tavakkoli@gmail.com>

  * contents/entries/194.php, contents/entries/195.php: ezafe kardan asami va
  file 194 va khabare namaloom 195

  * contents/entries/194.php, contents/entries/195.php: ezafe kardan asami va
  file 194 va khabare namaloom 195

2014-03-13  Behnam Tavakkoli kermani <b.tavakkoli@gmail.com>

  * photos/193/92_12_13-1_thumb.jpg, photos/193/92_12_13-2_thumb.jpg: ezafe
  kardan aks akhare jalase 193

  * photos/193/92_12_13-1.jpg, photos/193/92_12_13-1_thumb.jpg,
  photos/193/92_12_13-2.jpg, photos/193/92_12_13-2_thumb.jpg: ezafe kardan aks
  akhare jalase 193

2014-02-19  Behnam Tavakkoli kermani <b.tavakkoli@gmail.com>

  * photos/192/92_11_29.jpg, photos/192/92_11_29_thumb.png: ezafe kardane akse
  192

  * contents/entries/192.php, contents/entries/193.php: ezafe kardan asami 192
  + pish-khabar 193

2014-02-12  Behnam Tavakkoli kermani <b.tavakkoli@gmail.com>

  * contents/entries/191.php, contents/entries/192.php: ezafe kardan asami 191
  + file'ha + khabare 192

2014-02-01  Behnam Tavakkoli kermani <b.tavakkoli@gmail.com>

  * contents/entries/189.php, contents/entries/190.php,
  contents/entries/191.php: ezafe kardan asami 190 + khabare 191 + eslah link
  189

2014-01-17  Behnam Tavakkoli kermani <b.tavakkoli@gmail.com>

  * contents/entries/189.php, contents/entries/190.php: ezafe kardan asami +
  khabare 190

2014-01-04  Behnam Tavakkoli kermani <b.tavakkoli@gmail.com>

  * contents/entries/188.php, contents/entries/189.php: ezafe kardan khabare
  189 va asami 188?

2013-12-24  Behnam Tavakkoli kermani <b.tavakkoli@gmail.com>

  * contents/entries/184.php, contents/entries/186.php,
  contents/entries/187.php, contents/entries/188.php: ezafe kardane asami +
  darje khabare 188

2013-12-08  Behnam Tavakkoli kermani <b.tavakkoli@gmail.com>

  * contents/entries/186.php, contents/entries/187.php: ezafe kardan asami 186
  va khabare 187

2013-11-25  Behnam Tavakkoli kermani <b.tavakkoli@gmail.com>

  * contents/entries/185.php, contents/entries/186.php: ezafe kardane asami 185
  + khabare 186

2013-10-30  Behnam Tavakkoli kermani <b.tavakkoli@gmail.com>

  * contents/entries/184.php, contents/entries/185.php: darje asami

  * contents/entries/184.php, contents/entries/185.php: darje asami

  * contents/entries/183.php, contents/entries/184.php,
  contents/entries/185.php: ezafe kardan asami va khabar

2013-10-24  Behnam Tavakkoli kermani <b.tavakkoli@gmail.com>

  * images/map-tosan.jpg, style.css: eslahe neshani


jdf.php
<?php
function jdate($type, $maket = "now") {
	//set 1 if you want translate number to farsi or if you don't like set 0
	//$transnumber = 0;
	///chosse your timezone
	$TZhours = 3;
	$TZminute = 30;
	$need = "";
	$result1 = "";
	$result = "";
	if ($maket == "now") {
		$year = date ( "Y" );
		$month = date ( "m" );
		$day = date ( "d" );
		list ( $jyear, $jmonth, $jday ) = gregorian_to_jalali ( $year, $month, $day );
		$maket = mktime ( date ( "H" ) + $TZhours, date ( "i" ) + $TZminute, date ( "s" ), date ( "m" ), date ( "d" ), date ( "Y" ) );
	} else {
		//$maket=0;
		$maket += $TZhours * 3600 + $TZminute * 60;
		$date = date ( "Y-m-d", $maket );
		list ( $year, $month, $day ) = preg_split ( '/-/', $date );

		list ( $jyear, $jmonth, $jday ) = gregorian_to_jalali ( $year, $month, $day );
	}

	$need = $maket;
	$year = date ( "Y", $need );
	$month = date ( "m", $need );
	$day = date ( "d", $need );
	$i = 0;
	$subtype = "";
	$subtypetemp = "";
	list ( $jyear, $jmonth, $jday ) = gregorian_to_jalali ( $year, $month, $day );
	while ( $i < strlen ( $type ) ) {
		$subtype = substr ( $type, $i, 1 );
		if ($subtypetemp == "\\") {
			$result .= $subtype;
			$i ++;
			continue;
		}

		switch ($subtype) {

			case "A" :
				$result1 = date ( "a", $need );
				if ($result1 == "pm")
					$result .= "&#1576;&#1593;&#1583;&#1575;&#1586;&#1592;&#1607;&#1585;";
				else
					$result .= "&#1602;&#1576;&#1604;&#8207;&#1575;&#1586;&#1592;&#1607;&#1585;";
				break;

			case "a" :
				$result1 = date ( "a", $need );
				if ($result1 == "pm")
					$result .= "&#1576;&#46;&#1592;";
				else
					$result .= "&#1602;&#46;&#1592;";
				break;
			case "d" :
				if ($jday < 10)
					$result1 = "0" . $jday;
				else
					$result1 = $jday;
				if ($transnumber == 1)
					$result .= Convertnumber2farsi ( $result1 );
				else
					$result .= $result1;
				break;
			case "D" :
				$result1 = date ( "D", $need );
				if ($result1 == "Thu")
					$result1 = "&#1662;";
				else if ($result1 == "Sat")
					$result1 = "&#1588;";
				else if ($result1 == "Sun")
					$result1 = "&#1609;";
				else if ($result1 == "Mon")
					$result1 = "&#1583;";
				else if ($result1 == "Tue")
					$result1 = "&#1587;";
				else if ($result1 == "Wed")
					$result1 = "&#1670;";
				else if ($result1 == "Thu")
					$result1 = "&#1662;";
				else if ($result1 == "Fri")
					$result1 = "&#1580;";
				$result .= $result1;
				break;
			case "F" :
				$result .= monthname ( $jmonth );
				break;
			case "g" :
				$result1 = date ( "g", $need );
				if ($transnumber == 1)
					$result .= Convertnumber2farsi ( $result1 );
				else
					$result .= $result1;
				break;
			case "G" :
				$result1 = date ( "G", $need );
				if ($transnumber == 1)
					$result .= Convertnumber2farsi ( $result1 );
				else
					$result .= $result1;
				break;
			case "h" :
				$result1 = date ( "h", $need );
				if ($transnumber == 1)
					$result .= Convertnumber2farsi ( $result1 );
				else
					$result .= $result1;
				break;
			case "H" :
				$result1 = date ( "H", $need );
				if ($transnumber == 1)
					$result .= Convertnumber2farsi ( $result1 );
				else
					$result .= $result1;
				break;
			case "i" :
				$result1 = date ( "i", $need );
				if ($transnumber == 1)
					$result .= Convertnumber2farsi ( $result1 );
				else
					$result .= $result1;
				break;
			case "j" :
				$result1 = $jday;
				if ($transnumber == 1)
					$result .= Convertnumber2farsi ( $result1 );
				else
					$result .= $result1;
				break;
			case "l" :
				$result1 = date ( "l", $need );
				if ($result1 == "Saturday")
					$result1 = "شنبه";
				else if ($result1 == "Sunday")
					$result1 = "یک‌شنبه";
				else if ($result1 == "Monday")
					$result1 = "دو‌شنبه";
				else if ($result1 == "Tuesday")
					$result1 = "سه‌شنبه";
				else if ($result1 == "Wednesday")
					$result1 = "جهار‌شنبه";
				else if ($result1 == "Thursday")
					$result1 = "پنج‌شنبه";
				else if ($result1 == "Friday")
					$result1 = "جمعه";
				$result .= $result1;
				break;
			case "m" :
				if ($jmonth < 10)
					$result1 = "0" . $jmonth;
				else
					$result1 = $jmonth;
				if ($transnumber == 1)
					$result .= Convertnumber2farsi ( $result1 );
				else
					$result .= $result1;
				break;
			case "M" :
				$result .= short_monthname ( $jmonth );
				break;
			case "n" :
				$result1 = $jmonth;
				if ($transnumber == 1)
					$result .= Convertnumber2farsi ( $result1 );
				else
					$result .= $result1;
				break;
			case "s" :
				$result1 = date ( "s", $need );
				if ($transnumber == 1)
					$result .= Convertnumber2farsi ( $result1 );
				else
					$result .= $result1;
				break;
			case "S" :
				$result .= "";
				break;
			case "t" :
				$result .= lastday ( $month, $day, $year );
				break;
			case "w" :
				$result1 = date ( "w", $need );
				if ($transnumber == 1)
					$result .= Convertnumber2farsi ( $result1 );
				else
					$result .= $result1;
				break;
			case "y" :
				$result1 = substr ( $jyear, 2, 4 );
				if ($transnumber == 1)
					$result .= Convertnumber2farsi ( $result1 );
				else
					$result .= $result1;
				break;
			case "Y" :
				$result1 = $jyear;
				if ($transnumber == 1)
					$result .= Convertnumber2farsi ( $result1 );
				else
					$result .= $result1;
				break;
			case "U" :
				$result .= mktime ();
				break;
			case "Z" :
				$result .= days_of_year ( $jmonth, $jday, $jyear );
				break;
			case "L" :
				list ( $tmp_year, $tmp_month, $tmp_day ) = jalali_to_gregorian ( 1384, 12, 1 );
				echo $tmp_day;
				/*if(lastday($tmp_month,$tmp_day,$tmp_year)=="31")
					$result.="1";
				else
					$result.="0";
					*/
				break;
			default :
				$result .= $subtype;
		}
		$subtypetemp = substr ( $type, $i, 1 );
		$i ++;
	}
	return $result;
}


//Find Number Of Days In This Month
function lastday($month, $day, $year) {
	$jday2 = "";
	$jdate2 = "";
	$lastdayen = date ( "d", mktime ( 0, 0, 0, $month + 1, 0, $year ) );
	list ( $jyear, $jmonth, $jday ) = gregorian_to_jalali ( $year, $month, $day );
	$lastdatep = $jday;
	$jday = $jday2;
	while ( $jday2 != "1" ) {
		if ($day < $lastdayen) {
			$day ++;
			list ( $jyear, $jmonth, $jday2 ) = gregorian_to_jalali ( $year, $month, $day );
			if ($jdate2 == "1")
				break;
			if ($jdate2 != "1")
				$lastdatep ++;
		} else {
			$day = 0;
			$month ++;
			if ($month == 13) {
				$month = "1";
				$year ++;
			}
		}

	}
	return $lastdatep - 1;
}

//Find days in this year untile now
function days_of_year($jmonth, $jday, $jyear) {
	$year = "";
	$month = "";
	$year = "";
	$result = "";
	if ($jmonth == "01")
		return $jday;
	for($i = 1; $i < $jmonth || $i == 12; $i ++) {
		list ( $year, $month, $day ) = jalali_to_gregorian ( $jyear, $i, "1" );
		$result += lastday ( $month, $day, $year );
	}
	return $result + $jday;
}

//translate number of month to name of month
function monthname($month) {

	if ($month == "01")
		return "فروردین";

	if ($month == "02")
		return "اردیبهشت";

	if ($month == "03")
		return "خرداد";

	if ($month == "04")
		return "تیر";

	if ($month == "05")
		return "مرداد";

	if ($month == "06")
		return "شهریور";

	if ($month == "07")
		return "مهر";

	if ($month == "08")
		return "آبان";

	if ($month == "09")
		return "آذر";

	if ($month == "10")
		return "دی";

	if ($month == "11")
		return "بهمن";

	if ($month == "12")
		return "اسفند";
}

function short_monthname($month) {

	if ($month == "01")
		return "فروردین";

	if ($month == "02")
		return "اردیبهشت";

	if ($month == "03")
		return "خرداد";

	if ($month == "04")
		return "تیر";

	if ($month == "05")
		return "مرداد";

	if ($month == "06")
		return "شهریور";

	if ($month == "07")
		return "مهر";

	if ($month == "08")
		return "آبان";

	if ($month == "09")
		return "آذر";

	if ($month == "10")
		return "دی";

	if ($month == "11")
		return "بهمن";

	if ($month == "12")
		return "اسفند";
}

function jmaketime($hour,$minute,$second,$jmonth,$jday,$jyear)
{
	list( $year, $month, $day ) = jalali_to_gregorian($jyear, $jmonth, $jday);
	$i=mktime($hour,$minute,$second,$month,$day,$year);
	return $i;
}

////here convert to  number in persian
function Convertnumber2farsi($srting) {
	return str_replace(array(0,1,2,3,4,5,6,7,8,9), array('۰','۱','۲','۳','۴','۵','۶','۷','۸','۹'), $string);
} ///end conver to number in persian

// "jalali.php" is convertor to and from Gregorian and Jalali calendars.
// Copyright (C) 2000  Roozbeh Pournader and Mohammad Toossi
//
// This program is free software; you can redistribute it and/or
// modify it under the terms of the GNU General Public License
// as published by the Free Software Foundation; either version 2
// of the License, or (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// A copy of the GNU General Public License is available from:
//
//    <a href="http://www.gnu.org/copyleft/gpl.html" target="_blank">http://www.gnu.org/copyleft/gpl.html</a>
//


function div($a, $b) {
	return ( int ) ($a / $b);
}

function gregorian_to_jalali($g_y, $g_m, $g_d) {
	$g_days_in_month = array (31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31 );
	$j_days_in_month = array (31, 31, 31, 31, 31, 31, 30, 30, 30, 30, 30, 29 );

	$gy = $g_y - 1600;
	$gm = $g_m - 1;
	$gd = $g_d - 1;

	$g_day_no = 365 * $gy + div ( $gy + 3, 4 ) - div ( $gy + 99, 100 ) + div ( $gy + 399, 400 );

	for($i = 0; $i < $gm; ++ $i)
		$g_day_no += $g_days_in_month [$i];
	if ($gm > 1 && (($gy % 4 == 0 && $gy % 100 != 0) || ($gy % 400 == 0)))
      /* leap and after Feb */
      $g_day_no ++;
	$g_day_no += $gd;

	$j_day_no = $g_day_no - 79;

	$j_np = div ( $j_day_no, 12053 ); /* 12053 = 365*33 + 32/4 */
	$j_day_no = $j_day_no % 12053;

	$jy = 979 + 33 * $j_np + 4 * div ( $j_day_no, 1461 ); /* 1461 = 365*4 + 4/4 */

	$j_day_no %= 1461;

	if ($j_day_no >= 366) {
		$jy += div ( $j_day_no - 1, 365 );
		$j_day_no = ($j_day_no - 1) % 365;
	}

	for($i = 0; $i < 11 && $j_day_no >= $j_days_in_month [$i]; ++ $i)
		$j_day_no -= $j_days_in_month [$i];
	$jm = $i + 1;
	$jd = $j_day_no + 1;

	return array ($jy, $jm, $jd );
}

function jalali_to_gregorian($j_y, $j_m, $j_d) {
	$g_days_in_month = array (31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31 );
	$j_days_in_month = array (31, 31, 31, 31, 31, 31, 30, 30, 30, 30, 30, 29 );

	$jy = $j_y - 979;
	$jm = $j_m - 1;
	$jd = $j_d - 1;

	$j_day_no = 365 * $jy + div ( $jy, 33 ) * 8 + div ( $jy % 33 + 3, 4 );
	for($i = 0; $i < $jm; ++ $i)
		$j_day_no += $j_days_in_month [$i];

	$j_day_no += $jd;

	$g_day_no = $j_day_no + 79;

	$gy = 1600 + 400 * div ( $g_day_no, 146097 ); /* 146097 = 365*400 + 400/4 - 400/100 + 400/400 */
	$g_day_no = $g_day_no % 146097;

	$leap = true;
	if ($g_day_no >= 36525) /* 36525 = 365*100 + 100/4 */
   {
		$g_day_no --;
		$gy += 100 * div ( $g_day_no, 36524 ); /* 36524 = 365*100 + 100/4 - 100/100 */
		$g_day_no = $g_day_no % 36524;

		if ($g_day_no >= 365)
			$g_day_no ++;
		else
			$leap = false;
	}

	$gy += 4 * div ( $g_day_no, 1461 ); /* 1461 = 365*4 + 4/4 */
	$g_day_no %= 1461;

	if ($g_day_no >= 366) {
		$leap = false;

		$g_day_no --;
		$gy += div ( $g_day_no, 365 );
		$g_day_no = $g_day_no % 365;
	}

	for($i = 0; $g_day_no >= $g_days_in_month [$i] + ($i == 1 && $leap); $i ++)
		$g_day_no -= $g_days_in_month [$i] + ($i == 1 && $leap);
	$gm = $i + 1;
	$gd = $g_day_no + 1;

	return array ($gy, $gm, $gd );
}
?>